packages/protocol/contracts/L2/TaikoL2.sol

```diff
@@ -139,51 +160,125 @@ contract TaikoL2 is EssentialContract, TaikoL2Signer, ICrossChainSync {
         }
     }
 
+    /// @notice Returns EIP1559 related configurations
+    function getConfig() public pure virtual returns (Config memory config) {
+        config.blockGasTarget = 15 * 1e6 * 10; // 10x Ethereum gas target
+        config.basefeeAdjustmentQuotient = 8;
+        config.blockRewardPerL1Block = 1e15; // 0.001 Taiko token;
+        config.blockRewardPoolMax = 12e18; // 12 Taiko token
+        config.blockRewardPoolPctg = 40; // 40%
+    }
+
+    // In situations where the network lacks sufficient transactions for the
+    // proposer to profit, they are still obligated to pay the prover the
+    // proving fee, which can be a substantial cost compared to the total L2
+    // transaction fees collected. As a solution, Taiko mints additional Taiko
+    // tokens per second as block rewards. It's important to note that if
+    // multiple blocks are proposed within the same L1 block, only the first one
+    // will receive the block reward.
+    //
+    // The block reward doesn't undergo automatic halving; instead, we depend on
+    // Taiko DAO to make necessary adjustments to the rewards. uint96
+    // rewardBase;
+    //
+    // Reward block proposers with Taiko tokens to encourage chain adoption and
+    // ensure liveness. Rewards are issued only if `blockRewardPerL1Block` and
+    // `blockRewardPoolMax` are set to nonzero values in the configuration.
+    //
+    // Mint additional tokens into the reward pool as L1 block numbers increase,
+    // to incentivize future proposers.
+    function _rewardParentBlock(
+        Config memory config,
+        uint64 l1Height,
+        uint32 parentGasUsed
+    )
+        private
+        returns (uint128 blockReward)
+    {
+        if (
+            config.blockRewardPerL1Block == 0 || config.blockRewardPoolMax == 0
+                || config.blockRewardPoolPctg == 0 || latestSyncedL1Height == 0
+                || accumulatedReward == 0
+        ) return 0;
+
+        if (latestSyncedL1Height < l1Height) {
+            uint256 extraRewardMinted = uint256(l1Height - latestSyncedL1Height)
+                * config.blockRewardPerL1Block;
+
+            // Reward pool is capped to `blockRewardPoolMax`
+            accumulatedReward = uint128(
+                (extraRewardMinted + accumulatedReward).min(
+                    config.blockRewardPoolMax
+                )
+            );
+        }
+
+        if (avgGasUsed == 0) {
+            avgGasUsed = parentGasUsed;
+            return 0;
+        }
+
+        avgGasUsed = avgGasUsed / 1024 * 1023 + parentGasUsed / 1024;
+
+        uint128 maxBlockReward =
+            accumulatedReward / 100 * config.blockRewardPoolPctg;
+        accumulatedReward -= maxBlockReward;
+
+        if (
+            parentGasUsed <= ANCHOR_GAS_DEDUCT
+                || avgGasUsed <= ANCHOR_GAS_DEDUCT || parentProposer == address(0)
+        ) {
+            return 0;
+        }
+
+        address tt = resolve("taiko", true);
```

Comment:
```suggestion
        address tt = resolve("taiko_token", true);
```
The name `"taiko"` on L2 is used (in L2 signal service) which will be resolved to this `ProxiedTaikoL2` contract, so we may need to use another name here: 

```solidity
bytes32 syncedSignalRoot = ICrossChainSync(resolve("taiko", false))
            .getCrossChainSignalRoot(signalProof.height);
```